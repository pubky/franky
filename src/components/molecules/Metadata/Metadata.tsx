interface MetadataProps {
  title: string;
  description: string;
  image?: string;
  type?: string;
  url?: string;
  siteName?: string;
  locale?: string;
  author?: string;
  keywords?: string;
  robots?: boolean;
}

const DEFAULT_PREVIEW_IMAGE = '/preview.png';
const DEFAULT_SITE_NAME = 'Pubky App';
const DEFAULT_LOCALE = 'en_US';
const DEFAULT_AUTHOR = 'Pubky Team';
const DEFAULT_KEYWORDS = 'pubky, social media, decentralized, key, pkarr, pubky core';
const DEFAULT_TYPE = 'website';
const DEFAULT_URL = 'https://pubky.app';

export function Metadata({
  title,
  description,
  image,
  type,
  url,
  siteName,
  locale,
  author,
  keywords,
  robots,
}: MetadataProps) {
  return {
    title,
    description,
    keywords: keywords || DEFAULT_KEYWORDS,
    authors: [{ name: author || DEFAULT_AUTHOR }],
    creator: author || DEFAULT_AUTHOR,
    icons: {
      icon: '/pubky-favicon.svg',
      shortcut: '/pubky-favicon.svg',
      apple: '/pubky-favicon.svg',
    },
    openGraph: {
      title,
      description,
      url: url || DEFAULT_URL,
      siteName: siteName || DEFAULT_SITE_NAME,
      images: [
        {
          url: image || DEFAULT_PREVIEW_IMAGE,
          width: 1200,
          height: 630,
          alt: title,
        },
      ],
      locale: locale || DEFAULT_LOCALE,
      type: type || DEFAULT_TYPE,
    },
    twitter: {
      card: 'summary_large_image',
      title,
      description,
      images: [image || DEFAULT_PREVIEW_IMAGE],
      creator: '@pubkyapp',
      site: '@pubkyapp',
    },
    robots: {
      index: robots ?? true,
      follow: robots ?? true,
    },
    alternates: {
      canonical: url || DEFAULT_URL,
    },
  };
}
