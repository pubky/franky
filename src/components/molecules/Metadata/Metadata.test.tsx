import { describe, it, expect } from 'vitest';
import { Metadata } from './Metadata';

describe('Metadata - Snapshots', () => {
  it('matches snapshot for default metadata configuration', () => {
    const result = Metadata({
      title: 'Test Title',
      description: 'Test Description',
    });
    expect(result).toMatchSnapshot();
  });

  it('matches snapshot for metadata with custom parameters', () => {
    const result = Metadata({
      title: 'Custom Title',
      description: 'Custom Description',
      image: '/custom-image.jpg',
      type: 'article',
      url: 'https://custom-url.com',
      siteName: 'Custom Site',
      locale: 'it_IT',
      author: 'Custom Author',
      keywords: 'custom, keywords',
      robots: false,
    });
    expect(result).toMatchSnapshot();
  });

  it('matches snapshot for metadata with minimal configuration', () => {
    const result = Metadata({
      title: 'Minimal',
      description: 'Minimal description',
    });
    expect(result).toMatchSnapshot();
  });

  it('matches snapshot for metadata with image', () => {
    const result = Metadata({
      title: 'Image Test',
      description: 'Testing custom image',
      image: '/image.jpg',
    });
    expect(result).toMatchSnapshot();
  });

  it('matches snapshot with empty strings', () => {
    const result = Metadata({
      title: '',
      description: '',
    });
    expect(result).toMatchSnapshot();
  });

  it('matches snapshot with long title and description', () => {
    const longTitle = 'A'.repeat(1000);
    const longDescription = 'B'.repeat(1000);
    const result = Metadata({
      title: longTitle,
      description: longDescription,
    });
    expect(result).toMatchSnapshot();
  });

  it('matches snapshot with special characters', () => {
    const result = Metadata({
      title: 'Special chars: <>&"\'',
      description: 'More special: ©®™€£¥',
    });
    expect(result).toMatchSnapshot();
  });
});
