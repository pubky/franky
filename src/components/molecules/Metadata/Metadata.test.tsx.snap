// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Metadata - Snapshots > matches snapshot for default metadata configuration 1`] = `
{
  "alternates": {
    "canonical": "https://pubky.app",
  },
  "authors": [
    {
      "name": "Pubky Team",
    },
  ],
  "creator": "Pubky Team",
  "description": "Test Description",
  "icons": {
    "apple": "/pubky-favicon.svg",
    "icon": "/pubky-favicon.svg",
    "shortcut": "/pubky-favicon.svg",
  },
  "keywords": "pubky, social media, decentralized, key, pkarr, pubky core",
  "openGraph": {
    "description": "Test Description",
    "images": [
      {
        "alt": "Test Title",
        "height": 630,
        "url": "/preview.png",
        "width": 1200,
      },
    ],
    "locale": "en_US",
    "siteName": "Pubky App",
    "title": "Test Title",
    "type": "website",
    "url": "https://pubky.app",
  },
  "robots": {
    "follow": true,
    "index": true,
  },
  "title": "Test Title",
  "twitter": {
    "card": "summary_large_image",
    "creator": "@getpubky",
    "description": "Test Description",
    "images": [
      "/preview.png",
    ],
    "site": "https://pubky.app",
    "title": "Test Title",
  },
}
`;

exports[`Metadata - Snapshots > matches snapshot for metadata with custom parameters 1`] = `
{
  "alternates": {
    "canonical": "https://custom-url.com",
  },
  "authors": [
    {
      "name": "Custom Author",
    },
  ],
  "creator": "Custom Author",
  "description": "Custom Description",
  "icons": {
    "apple": "/pubky-favicon.svg",
    "icon": "/pubky-favicon.svg",
    "shortcut": "/pubky-favicon.svg",
  },
  "keywords": "custom, keywords",
  "openGraph": {
    "description": "Custom Description",
    "images": [
      {
        "alt": "Custom Title",
        "height": 630,
        "url": "/custom-image.jpg",
        "width": 1200,
      },
    ],
    "locale": "it_IT",
    "siteName": "Custom Site",
    "title": "Custom Title",
    "type": "article",
    "url": "https://custom-url.com",
  },
  "robots": {
    "follow": false,
    "index": false,
  },
  "title": "Custom Title",
  "twitter": {
    "card": "summary_large_image",
    "creator": "@getpubky",
    "description": "Custom Description",
    "images": [
      "/custom-image.jpg",
    ],
    "site": "https://pubky.app",
    "title": "Custom Title",
  },
}
`;

exports[`Metadata - Snapshots > matches snapshot for metadata with image 1`] = `
{
  "alternates": {
    "canonical": "https://pubky.app",
  },
  "authors": [
    {
      "name": "Pubky Team",
    },
  ],
  "creator": "Pubky Team",
  "description": "Testing custom image",
  "icons": {
    "apple": "/pubky-favicon.svg",
    "icon": "/pubky-favicon.svg",
    "shortcut": "/pubky-favicon.svg",
  },
  "keywords": "pubky, social media, decentralized, key, pkarr, pubky core",
  "openGraph": {
    "description": "Testing custom image",
    "images": [
      {
        "alt": "Image Test",
        "height": 630,
        "url": "/image.jpg",
        "width": 1200,
      },
    ],
    "locale": "en_US",
    "siteName": "Pubky App",
    "title": "Image Test",
    "type": "website",
    "url": "https://pubky.app",
  },
  "robots": {
    "follow": true,
    "index": true,
  },
  "title": "Image Test",
  "twitter": {
    "card": "summary_large_image",
    "creator": "@getpubky",
    "description": "Testing custom image",
    "images": [
      "/image.jpg",
    ],
    "site": "https://pubky.app",
    "title": "Image Test",
  },
}
`;

exports[`Metadata - Snapshots > matches snapshot for metadata with minimal configuration 1`] = `
{
  "alternates": {
    "canonical": "https://pubky.app",
  },
  "authors": [
    {
      "name": "Pubky Team",
    },
  ],
  "creator": "Pubky Team",
  "description": "Minimal description",
  "icons": {
    "apple": "/pubky-favicon.svg",
    "icon": "/pubky-favicon.svg",
    "shortcut": "/pubky-favicon.svg",
  },
  "keywords": "pubky, social media, decentralized, key, pkarr, pubky core",
  "openGraph": {
    "description": "Minimal description",
    "images": [
      {
        "alt": "Minimal",
        "height": 630,
        "url": "/preview.png",
        "width": 1200,
      },
    ],
    "locale": "en_US",
    "siteName": "Pubky App",
    "title": "Minimal",
    "type": "website",
    "url": "https://pubky.app",
  },
  "robots": {
    "follow": true,
    "index": true,
  },
  "title": "Minimal",
  "twitter": {
    "card": "summary_large_image",
    "creator": "@getpubky",
    "description": "Minimal description",
    "images": [
      "/preview.png",
    ],
    "site": "https://pubky.app",
    "title": "Minimal",
  },
}
`;

exports[`Metadata - Snapshots > matches snapshot with empty strings 1`] = `
{
  "alternates": {
    "canonical": "https://pubky.app",
  },
  "authors": [
    {
      "name": "Pubky Team",
    },
  ],
  "creator": "Pubky Team",
  "description": "",
  "icons": {
    "apple": "/pubky-favicon.svg",
    "icon": "/pubky-favicon.svg",
    "shortcut": "/pubky-favicon.svg",
  },
  "keywords": "pubky, social media, decentralized, key, pkarr, pubky core",
  "openGraph": {
    "description": "",
    "images": [
      {
        "alt": "",
        "height": 630,
        "url": "/preview.png",
        "width": 1200,
      },
    ],
    "locale": "en_US",
    "siteName": "Pubky App",
    "title": "",
    "type": "website",
    "url": "https://pubky.app",
  },
  "robots": {
    "follow": true,
    "index": true,
  },
  "title": "",
  "twitter": {
    "card": "summary_large_image",
    "creator": "@getpubky",
    "description": "",
    "images": [
      "/preview.png",
    ],
    "site": "https://pubky.app",
    "title": "",
  },
}
`;

exports[`Metadata - Snapshots > matches snapshot with long title and description 1`] = `
{
  "alternates": {
    "canonical": "https://pubky.app",
  },
  "authors": [
    {
      "name": "Pubky Team",
    },
  ],
  "creator": "Pubky Team",
  "description": "BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB",
  "icons": {
    "apple": "/pubky-favicon.svg",
    "icon": "/pubky-favicon.svg",
    "shortcut": "/pubky-favicon.svg",
  },
  "keywords": "pubky, social media, decentralized, key, pkarr, pubky core",
  "openGraph": {
    "description": "BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB",
    "images": [
      {
        "alt": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
        "height": 630,
        "url": "/preview.png",
        "width": 1200,
      },
    ],
    "locale": "en_US",
    "siteName": "Pubky App",
    "title": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
    "type": "website",
    "url": "https://pubky.app",
  },
  "robots": {
    "follow": true,
    "index": true,
  },
  "title": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
  "twitter": {
    "card": "summary_large_image",
    "creator": "@getpubky",
    "description": "BBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB",
    "images": [
      "/preview.png",
    ],
    "site": "https://pubky.app",
    "title": "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA",
  },
}
`;

exports[`Metadata - Snapshots > matches snapshot with special characters 1`] = `
{
  "alternates": {
    "canonical": "https://pubky.app",
  },
  "authors": [
    {
      "name": "Pubky Team",
    },
  ],
  "creator": "Pubky Team",
  "description": "More special: ©®™€£¥",
  "icons": {
    "apple": "/pubky-favicon.svg",
    "icon": "/pubky-favicon.svg",
    "shortcut": "/pubky-favicon.svg",
  },
  "keywords": "pubky, social media, decentralized, key, pkarr, pubky core",
  "openGraph": {
    "description": "More special: ©®™€£¥",
    "images": [
      {
        "alt": "Special chars: <>&\\"'",
        "height": 630,
        "url": "/preview.png",
        "width": 1200,
      },
    ],
    "locale": "en_US",
    "siteName": "Pubky App",
    "title": "Special chars: <>&\\"'",
    "type": "website",
    "url": "https://pubky.app",
  },
  "robots": {
    "follow": true,
    "index": true,
  },
  "title": "Special chars: <>&\\"'",
  "twitter": {
    "card": "summary_large_image",
    "creator": "@getpubky",
    "description": "More special: ©®™€£¥",
    "images": [
      "/preview.png",
    ],
    "site": "https://pubky.app",
    "title": "Special chars: <>&\\"'",
  },
}
`;
